website/blog/2024-06-26-llm-as-judge/index.md

```diff
@@ -0,0 +1,522 @@
+            output="Our company supports flexible work arrangements.",
+            score=3,
+            justification="The answer is related to the context but does not specifically answer the question about the remote work policy."
+        ),
+        mlflow.metrics.genai.EvaluationExample(
+            input="""Question: What is the company's annual leave policy?
+context: "Employees are entitled to 20 days of paid annual leave per year. Leave must be approved by the employee's direct supervisor and should be planned in advance to ensure minimal disruption to work." """,
+            output="Employees are entitled to 20 days of paid annual leave per year, which must be approved by their supervisor.",
+            score=5,
+            justification="The answer is accurate and directly related to the question and context provided."
+        )]
+
+#  Define the custom metric
+faithfulness = mlflow.metrics.genai.make_genai_metric(
+    name="faithfulness",
+    definition="Assesses how well the answer relates to the question and provided context.",
+    grading_prompt="Score from 1-5, where 1 is not related at all and 5 is highly relevant and accurate.",
+    examples=examples)
+
+```
+
+Define out LLM function (in this case it can be any function that follows certain input /output formats that mlflow evalaute supports)
+
+```python
+# Using custom function
+def my_llm(inputs):
+    answers = []
+    system_prompt = "Please answer the following question in formal language based on the context provided."
+    for index, row in inputs.iterrows():
+        print('INPUTS:', row)
+        completion = openai.chat.completions.create(
+            model="gpt-3.5-turbo",
+            messages=[
+                {"role": "system", "content": system_prompt},
+                {"role": "user", "content": f"{row}"},
+            ],
+        )
+        answers.append(completion.choices[0].message.content)
+
+    return answers
+```
+
+Then our evaluation is similar to what we did before
+
+```python
+with mlflow.start_run() as run:
+    results = mlflow.evaluate(
+        my_llm,
+        eval_data,
+        model_type="text",
+        evaluators="default",
+        extra_metrics=[faithfulness],
+        evaluator_config={
+        "col_mapping": {
+            "inputs": "llm_inputs",
+           }}
+    )
+mlflow.end_run()
+```
+
+### GenAI Metrics
+
+As an alternative we can use mlflow built in metrics for genai using the same examples.
+
+MLflow offers a few pre-built metrics which uses LLM as the judge. Despite the difference under the hood, the usage is the same - put these metrics in the extra_metrics argument in mlflow.evaluate(). Here is the list of pre-built metrics:
+
+1. **[mlflow.metrics.genai.answer_similarity()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.answer_similarity)**
```

Comment:
Let's not do the highlight search results links (it's really distracting when following the link) and just link to the h<n> section of the docs page?

Reply:
I've udpated this, not sure if I totally understood what you meant but I've removed the bulletpoints and refered to the docs for the most part